src/kqueue.c

```diff
@@ -93,7 +93,7 @@ static int kq_poll(void* state, int timeout)
 	struct kevent events[16];
 	size_t max_events = sizeof(events) / sizeof(events[0]);
 
-	int nfds = kevent(self->fd, NULL, 0, events, max_events, &ts);
+	int nfds = kevent(self->fd, NULL, 0, events, max_events, timeout >= 0 ? &ts : NULL);
```

Comment:
This needs to be better explained in the commit message.

Reply:
Updated, commit message, clear now?

Reply:
It's better.

I like commit messages to be written as described here: https://github.com/any1/wayvnc/blob/master/CONTRIBUTING.md#commit-messages